sqlchain/block_test.go

```diff
@@ -17,56 +17,158 @@
 package sqlchain
 
 import (
-	"math/rand"
+	"math/big"
 	"reflect"
 	"testing"
+
+	"github.com/thunderdb/ThunderDB/crypto/asymmetric"
 )
 
 func TestSign(t *testing.T) {
 	block, err := createRandomBlock(rootHash, true)
 
 	if err != nil {
-		t.Fatalf("Error occurred: %s", err.Error())
+		t.Fatalf("Error occurred: %v", err)
```

Comment:
t.Fatalf("Error occurred: %s", err) should be better

Reply:
According to https://golang.org/pkg/fmt/, while processing Interface type:

4. If an operand implements the error interface, the Error method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any). 

%s and %v both should be fine. Approve plz.